test/tuple.cpp

```diff
@@ -416,6 +416,28 @@ TEST(CampTaggedTuple, MakeTagged)
   ASSERT_EQ(camp::get<s1>(t), 15);
 }
 
+template<typename Param>
+struct SearchType {};
+
+template<typename Param>
+struct OtherType {};
+
+template<typename T>
+struct IsSearchType : std::false_type {};
+
+template<typename Param>
+struct IsSearchType<SearchType<Param>> : std::true_type {};
+
+TEST(CampFilterByTypeTrait, MakeFilteredTuple)
+{
+  using BaseTupleType = camp::tuple<double, SearchType<int>, int, OtherType<int>, camp::tuple<double>, SearchType<int>>;
```

Comment:
Do you want to add some more tests like empty tuple, one element tuple, search type in the left most, in the right most, etc?